File: src/cdisc_library_api/specs/export_cdash_class_variables_row.clj
(ns cdisc-library-api.specs.export-cdash-class-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-cdash-class-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :Class) string?
   (ds/opt :Domain) string?
   (ds/opt :VariableOrder) string?
   (ds/opt :CDASHVariable) string?
   (ds/opt :CDASHVariableLabel) string?
   (ds/opt :DRAFTCDASHDefinition) string?
   (ds/opt :DomainSpecific) string?
   (ds/opt :QuestionText) string?
   (ds/opt :Prompt) string?
   (ds/opt :Type) string?
   (ds/opt :SDTMTarget) (s/coll-of string?)
   (ds/opt :MappingInstructions) string?
   (ds/opt :ControlledTerminologyCodelistName) string?
   (ds/opt :ImplementationNotes) string?
   })

(def export-cdash-class-variables-row-spec
  (ds/spec
    {:name ::export-cdash-class-variables-row
     :spec export-cdash-class-variables-row-data}))

File: src/main/scala/org/openapitools/client/api/ExportCdashClassVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportCdashClassVariablesRow._

case class ExportCdashClassVariablesRow (
  version: Option[String],
`class`: Option[String],
domain: Option[String],
variableOrder: Option[String],
cDASHVariable: Option[String],
cDASHVariableLabel: Option[String],
dRAFTCDASHDefinition: Option[String],
domainSpecific: Option[String],
questionText: Option[String],
prompt: Option[String],
`type`: Option[String],
sDTMTarget: Option[List[String]],
mappingInstructions: Option[String],
controlledTerminologyCodelistName: Option[String],
implementationNotes: Option[String])

object ExportCdashClassVariablesRow {
  import DateTimeCodecs._

  implicit val ExportCdashClassVariablesRowCodecJson: CodecJson[ExportCdashClassVariablesRow] = CodecJson.derive[ExportCdashClassVariablesRow]
  implicit val ExportCdashClassVariablesRowDecoder: EntityDecoder[ExportCdashClassVariablesRow] = jsonOf[ExportCdashClassVariablesRow]
  implicit val ExportCdashClassVariablesRowEncoder: EntityEncoder[ExportCdashClassVariablesRow] = jsonEncoderOf[ExportCdashClassVariablesRow]
}
